packages/azure-kusto-data/src/connectionBuilder.browser.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { DeviceCodeInfo } from "@azure/identity";
+import KustoConnectionStringBuilderBase from "./connectionBuilderBase";
+
+export class KustoConnectionStringBuilder extends KustoConnectionStringBuilderBase {
```

Comment:
as far as I know the only auth you're going to be able to get to work in a SPA is redirecting through AAD's OAuth endpoint

Reply:
The only credential that can be used in the browser is `InteractiveBrowserCredential`(offered by `@azure/identity`) with either popup or redirect mechanisms.